src/charm.py

```diff
@@ -622,6 +634,104 @@ def _get_cpu_extensions() -> list[str]:
                 del cpu_flags[0]
         return cpu_flags
 
+    def _get_access_nad_config(self) -> Dict[Any, Any]:
+        """Get access interface NAD config.
+
+        Returns:
+            config (dict): Access interface NAD config
+
+        """
+        config = {
+            "cniVersion": "0.3.1",
+            "ipam": {
+                "type": "static",
+                "routes": [
+                    {
+                        "dst": self._get_gnb_subnet_config(),
+                        "gw": self._get_access_network_gateway_ip_config(),
+                    },
+                ],
+                "addresses": [
+                    {
+                        "address": self._get_access_network_ip_config(),
+                    }
+                ],
+            },
+            "capabilities": {"mac": True},
+        }
+        if access_mtu := self._get_access_interface_mtu_config():
+            config.update({"mtu": access_mtu})
+        return config
+
+    def _get_core_nad_config(self) -> Dict[Any, Any]:
+        """Get core interface NAD config.
+
+        Returns:
+            config (dict): Core interface NAD config
+
+        """
+        config = {
+            "cniVersion": "0.3.1",
+            "ipam": {
+                "type": "static",
+                "addresses": [
+                    {
+                        "address": self._get_core_network_ip_config(),
+                    }
+                ],
+            },
+            "capabilities": {"mac": True},
+        }
+        if core_mtu := self._get_core_interface_mtu_config():
+            config.update({"mtu": core_mtu})
+        return config
+
+    def _get_core_interface_mtu_config(self) -> Optional[Any]:
+        """Get Core interface MTU size.
+
+        Returns:
+            mtu_size (int/None): If MTU size is not configured return None
+                                    If it is set, returns the configured value expected in int
+
+        """
+        return self.model.config.get("core-interface-mtu-size")
```

Comment:
If we want this to return an `int`, we can explicitly do the conversion here and set the return type:

```suggestion
    def _get_core_interface_mtu_config(self) -> Optional[int]:
        """Get Core interface MTU size.

        Returns:
            mtu_size (int/None): If MTU size is not configured return None
                                    If it is set, returns the configured value expected in int

        """
        if mtu_size := self.model.config.get("core-interface-mtu-size"):
            return int(mtu_size)
        return None
```

Otherwise, we should set the typing explicitly to `Optional[str]` and do the explicit conversion when we need it as an `int`. 

I think I prefer a conversion to `int` only when we do the validation.

Reply:
Returned as str, then str -> int validation is done in validation.